src/commonMain/kotlin/dev/romainguy/kotlin/math/Vector.kt

```diff
@@ -567,6 +567,23 @@ inline operator fun Float.minus(v: Float2) = Float2(this - v.x, this - v.y)
 inline operator fun Float.times(v: Float2) = Float2(this * v.x, this * v.y)
 inline operator fun Float.div(v: Float2) = Float2(this / v.x, this / v.y)
 
+inline fun compare(a: Float, b: Float, eps: Float? = null): Float = when {
```

Comment:
Use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-float/compare-to.html instead. Or at least write a version that supports NaN properly:

```
        if (a > b) return 1
        if (a < b) return -1

        return a.toBits().compareTo(b.toBits())
```